React/UIUtils/RCTUIUtils.m

```diff
@@ -21,6 +21,24 @@ RCTDimensions RCTGetDimensions(CGFloat fontScale)
   result.window = dims;
   result.screen = dims;
 
+  if (@available(iOS 11.0, *)) {
+    UIEdgeInsets windowInsets = RCTKeyWindow().rootViewController.view.safeAreaInsets;
```

Comment:
@janicduplessis why aren't you using `RCTKeyWindow().safeAreaInsets`? It does not force loading the view.

Reply:
I tries using that but the insets were always 0.

Reply:
Oh that is strange. Just to verify you are not running the code in a app extension right? Because RCTSharedApplication is returning nil when RCTRunningInAppExtension returns true. 

As an example we are using this is a large scale production app:

```objc
- (NSDictionary *) constantsToExport {
    return @{
             @"SAFE_AREA_INSETS_TOP": @([self safeAreaInsetsTop]),
             @"SAFE_AREA_INSETS_BOTTOM": @([self safeAreaInsetsBottom])
             };
}

- (CGFloat) safeAreaInsetsTop {
    if (@available(iOS 11.0, *)) {
        return UIApplication.sharedApplication.keyWindow.safeAreaInsets.top;
    }
    return 0;
}

- (CGFloat) safeAreaInsetsBottom {
    if (@available(iOS 11.0, *)) {
        return UIApplication.sharedApplication.keyWindow.safeAreaInsets.bottom;
    }
    return 0;
}
```

I wanted to open a PR myself but awesome that you are doing this already. thanks!

Reply:
Weird, I can try to investigate this further. I agree it would be cleaner to user keyWindow.safeAreaInsets.